diff --git a/quiche/Cargo.toml b/quiche/Cargo.toml
index bb21b15d..aba2f07f 100644
--- a/quiche/Cargo.toml
+++ b/quiche/Cargo.toml
@@ -33,6 +33,9 @@ default = ["boringssl-vendored"]
 # Build the vendored BoringSSL library.
 boringssl-vendored = []
 
+# use boringssl-wasm
+boringssl-wasm = []
+
 # Use the BoringSSL library provided by the boring crate.
 boringssl-boring-crate = ["boring", "foreign-types-shared"]
 
diff --git a/quiche/src/build.rs b/quiche/src/build.rs
index 1aef2e63..91d3a693 100644
--- a/quiche/src/build.rs
+++ b/quiche/src/build.rs
@@ -216,9 +223,9 @@ fn target_dir_path() -> std::path::PathBuf {
 }
 
 fn main() {
-    if cfg!(feature = "boringssl-vendored") &&
-        !cfg!(feature = "boringssl-boring-crate") &&
-        !cfg!(feature = "openssl")
+    if cfg!(feature = "boringssl-vendored")
+        && !cfg!(feature = "boringssl-boring-crate")
+        && !cfg!(feature = "openssl")
     {
         let bssl_dir = std::env::var("QUICHE_BSSL_PATH").unwrap_or_else(|_| {
             let mut cfg = get_boringssl_cmake_config();
@@ -261,6 +268,11 @@ fn main() {
         println!("cargo:rustc-cdylib-link-arg=-Wl,-undefined,dynamic_lookup");
     }
 
+    #[cfg(feature = "boringssl-wasm")]
+    {
+        println!("cargo:rustc-link-search=/home/runner/work/quiche/quiche/boringssl/build/");
+    }
+
     #[cfg(feature = "openssl")]
     {
         let pkgcfg = pkg_config::Config::new();
